lease/lessor.go

```diff
@@ -85,7 +100,10 @@ func (le *lessor) Revoke(id uint64) error {
 
 	delete(le.leaseMap, l.id)
 
-	// TODO: remove attached items
+	for item := range l.itemSet {
+		le.KV.DeleteRange([]byte(item.key), []byte(item.endRange))
```

Comment:
You can even lease on non existing key and range. This is by design and solves a lot of real problems in practice. If you cannot lease on non-existing key, there will be a race between creating key and creating lease. If one dies before creating the lease, then the key will never be removed.